engine/engine.go

```diff
@@ -435,14 +436,18 @@ func (e *Engine) GetLatestAdv(ctx context.Context) (cid.Cid, *schema.Advertiseme
 func (e *Engine) publishAdvForIndex(ctx context.Context, p peer.ID, addrs []multiaddr.Multiaddr, contextID []byte, md metadata.Metadata, isRm bool) (cid.Cid, error) {
```

Comment:
I think we could simplify the logic with variables and functions:
1. `isRemoveAllAds(contextID, isRm) bool, error` would return true if it's a message of that type, and false otherwise. It would also return an error if contextID is nil or has len 0 but isRm is false (`only remove messages can have empty context id`)
2. We could also have a `getAdType()` function that would return a simple struct saying the type of the advertisement and use a switch statement to avoid nested if statements